src/extension.ts

```diff
@@ -4,148 +4,194 @@
+    };
+
+    // Options to control the language client
+    let clientOptions: LanguageClientOptions = {
+        // Register the server for plain text documents
+        documentSelector: [{ scheme: 'file', language: 'smithy' }, { scheme: 'smithyjar', language: 'smithy' }],
+        synchronize: {
+            // Notify the server about file changes to 'smithy-build.json' or 'smithy.json' files contained in the workspace
+            fileEvents: workspace.createFileSystemWatcher('**/{smithy-build}.json')
+        }
+    };
+
+    // Create the language client and start the client.
+
+    client = new LanguageClient(
+        'smithyLsp',
+        'Smithy LSP',
+        createServer,
+        clientOptions
+    );
+    const smithyContentProvider = createSmithyContentProvider(client);
+    context.subscriptions.push(workspace.registerTextDocumentContentProvider('smithyjar', smithyContentProvider));
+
+    // Start the client. This will also launch the server
+    client.start();
 }
 
 export function deactivate(): Thenable<void> | undefined {
-	if (!client) {
-		return undefined;
-	}
-	return client.stop();
+    if (!client) {
+        return undefined;
+    }
+    return client.stop();
 }
 
 function createSmithyContentProvider(languageClient: LanguageClient): vscode.TextDocumentContentProvider {
-	return <vscode.TextDocumentContentProvider>{
-		provideTextDocumentContent: async (uri: vscode.Uri, token: CancellationToken): Promise<string> => {
-			return languageClient.sendRequest(ClassFileContentsRequest.type, { uri: uri.toString() }, token).then((v: string): string => {
-				return v || '';
-			});
-		}
-	}
+    return <vscode.TextDocumentContentProvider>{
+        provideTextDocumentContent: async (uri: vscode.Uri, token: CancellationToken): Promise<string> => {
+            return languageClient.sendRequest(ClassFileContentsRequest.type, { uri: uri.toString() }, token).then((v: string): string => {
+                return v || '';
+            });
+        }
+    }
+}
+
+function getCoursier(extensionPath: string): Promise<string> {
+    function binPath(filename: string) { 
+        return path.join(extensionPath, filename);
+    }
+    const urls = {
+        darwin: "https://github.com/coursier/coursier/releases/download/v2.0.6/cs-x86_64-apple-darwin",
+        linux: "https://github.com/coursier/coursier/releases/download/v2.0.6/cs-x86_64-pc-linux",
+        win32: "https://github.com/coursier/coursier/releases/download/v2.0.6/cs-x86_64-pc-win32.exe"
+    }
+    const targets = {
+        darwin: binPath("coursier"),
+        linux: binPath("coursier"),
+        win32: binPath("coursier.exe")
+    }
+    return downloadFile(urls[process.platform], targets[process.platform])
+}
+
+function downloadFile(url: string, targetFile: string): Promise<string> {
+    function get(url: string, resolve: (res: string) => void, reject: (err: Error) => void) {
+        https.get(url, response => {

```

Comment:
**suggestion (if-minor):** I would change the structure of this to be more promise-friendly:

1. promisify `https.get` - the point is to reduce the scope where `resolve`/`reject` are needed
2. remove `resolve`/`reject` from the parameters of `get`, removing the need for `new Promise` on the top level of `downloadFile`
3. use `.then` and other combinators on the promise returned from the new `https.get`, throw exceptions instead of calling `reject`, etc.

Reply:
done